### PR TITLE
feat: add a package for React components library

### PR DESCRIPTION
**Pull request by lauthieb**
## Pull Request checklist

🚨 Please review the [guidelines for contributing](../blob/main/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Don't request your main!
- [x] Make sure you are making a pull request against the **main branch** (left side).
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Description

Add a React components library.

## Does this introduce a breaking change?

- No

❤️ Thank you!

### CODE REVIEW

File: packages/showcases/web-components/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "@storybook/addon-links": "^6.1.11",
     "@storybook/web-components": "^6.1.11",
     "@vtmn/web-components": "*",
-    "babel-loader": "^8.2.2",
+    "babel-loader": "8.1.0",
```
Comment (gabbloquet) [inline packages/showcases/web-components/package.json:23]: You have to downgrade for a compliance that's it ?

File: packages/sources/react/rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  plugins: [
+    typescript({ tsconfig: target.tsconfig }),
+    external(),
+    resolve(),
+    url({ exclude: ['**/*.svg'] }),
+    svgr(),
+    postcss(),
+    commonjs(),
+  ],
+}));
```
Comment (gabbloquet) [inline packages/sources/react/rollup.config.js:34]: beautiful

File: packages/sources/react/src/react-app-env.d.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (gabbloquet) [inline packages/sources/react/src/react-app-env.d.ts:1]: is that file useful ?

**Review by gabbloquet — APPROVED** [review]
Amazing work, thanks a lot !

File: packages/showcases/web-components/package.json (hunk 1 above)
Comment (lauthieb) [reply packages/showcases/web-components/package.json:23]: Yes and alignment through babel loaders in other packages...

File: packages/sources/react/src/react-app-env.d.ts (hunk 3 above)
Comment (lauthieb) [reply packages/sources/react/src/react-app-env.d.ts:1]: Good question, I will check!

Comment (NicolasBreunin) [reply packages/sources/react/src/react-app-env.d.ts:1]: This file references the types of react-scripts, and helps with things like allowing for SVG imports :

[source](https://www.sitepoint.com/react-with-typescript-best-practices/#:~:text=json%20is%20the%20TypeScript%20configuration,like%20allowing%20for%20SVG%20imports.)

**Review by NicolasBreunin — APPROVED** [review]
great ! 👍

File: packages/sources/react/src/react-app-env.d.ts (hunk 3 above)
Comment (lauthieb) [reply packages/sources/react/src/react-app-env.d.ts:1]: Oh okay thanks @NicolasBreunin, we will keep it so.

**Review by Anthony-dkt — APPROVED** [review]

**Review by LionelSpanneut — APPROVED** [review]

File: packages/sources/react/src/components/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  return (
+    <button
+      className={`vtmn-btn ${className} vtmn-btn_variant--${variant} vtmn-btn_size--${size}`}
+      {...props}>
+      {children}
+    </button>
+  );
+};
+
+export default React.memo(Button);
```
Comment (sidbentifraouineatdecathlon) [inline packages/sources/react/src/components/Button/Button.tsx:45]: Nice ! ❤️

**Review by sidbentifraouineatdecathlon — APPROVED** [review]

**Comment by amen-souissi** [comment]
This is the source code of dktunited/vitamin-web. I think the contributors should be cited.

**Pull request merged**